lru-cache.go

```diff
@@ -0,0 +1,107 @@
+package main
+
+import "fmt"
+
+type DLLNode struct {
+	key, value int
+	next, prev *DLLNode
+}
+
+// constructor for DLL
+func NewDLLNode(key, value int) *DLLNode {
+	return &DLLNode{key: key, value: value}
+}
+
+type LRUCache struct {
+	capacity int
+	count    int
+	head     *DLLNode
+	tail     *DLLNode
+	mp       map[int]*DLLNode
+}
+
+// constructor for LRUCache
+func NewLRUCache(cap int) *LRUCache {
+	// sentinel head and tail nodes to avoid handling corner cases
+	head := NewDLLNode(0, 0)
```

Comment:
head and tail can also be simple pointer to DLL. But to avoid handling corner cases I am using head and tail as sentinel nodes.